README.md

```diff
@@ -185,45 +194,313 @@ interface LookerEmbedUser {
+- `api_token` - this token is used for api calls. This token lives for 10 minutes.
+
+A time to live for each token is also returned for each token. It is important that the response of the `acquire_embed_cookieless_session` be returned to the browser with the exception of the `session_reference_token`.
+
+The example below uses javascript, but you may utilize the Looker SDK in the language of your choice. Note that this is very simplistic implementation for demonstration purposes only. An actual implementation should be a lot more robust.
+
+```javascript
+// Simple endpoint to acquire a user session. In this case the user has been
+// is pulled from a configuration file. In a real life application the user
+// would be derived from a session.
+app.get('/acquire-embed-session', async function (req, res) {
+  try {
+    const tokens = await acquireEmbedSession(req.headers['user-agent'], user)
+    res.json(tokens)
+  } catch (err) {
+    res.status(400).send({ message: err.message })
+  }
+})
+
+// The Looker session. In a real application this should not be a global variable,
+let lookerSession
+
+// A very simple cache for storing embed sessions. In a real life application the
+// embed session should be stored with the applications users session.
+const embedSessions = {}
+
+// Simple function to acquire a looker session and then acquire an embed
+// session.
+async function acquireEmbedSession(userAgent, user) {
+  await acquireLookerSession()
+  return acquireEmbedSessionInternal(userAgent, user)
+}
+
+// Simple function to acquire a Looker session.
+const acquireLookerSession = async () => {
+  if (!lookerSession || !lookerSession.activeToken.isActive()) {
+    const { api_url, client_id, client_secret, verify_ssl } = config
+    try {
+      const lookerSettings = DefaultSettings()
+      lookerSettings.readConfig = () => {
+        return {
+          client_id,
+          client_secret,
+        }
+      }
+      lookerSettings.base_url = api_url
+      lookerSettings.verify_ssl = verify_ssl
+      lookerSession = new NodeSession(lookerSettings)
+      lookerSession.login()
+    } catch (error) {
+      console.error('login failed', { error })
+      throw error
+    }
+  }
+}
+
+// Simple function to acquire the embed session.
+// Note as an additional layer of security the user agent of users
+// browser is associated with the embed session. It is important
+// that this is available when the embed session is created.
+const acquireEmbedSessionInternal = async (userAgent, user) => {
+  try {
+    const cacheKey = `${user.external_user_id}/${userAgent}`
+    const embedSession = embedSessions[cacheKey]
+    const request = {
+      ...user,
+      session_reference_token: embedSession?.session_reference_token,
+    }
+    const sdk = new Looker40SDK(lookerSession)
+    const response = await sdk.ok(
+      sdk.acquire_embed_cookieless_session(request, {
```

Comment:
Do you think it's worth using functional SDK syntax here instead?

Reply:
As per discussion, has no benefit in node and is more discoverable. Leaving as is for now